lib/Controller/SettingsController.php

```diff
@@ -297,6 +298,14 @@ public function setPersonalSettings($templateFolder) {
 				$status = 'error';
 			}
 		}
+		if ($zoteroAPIKeyInput !== null) {
+			try {
+				$this->config->setUserValue($this->userId, 'richdocuments', 'zoteroAPIKey', $zoteroAPIKeyInput);
```

Comment:
Is there a way we could verify that the API key is valid here already? Even if it is only used by online?

Reply:
it can be done, but then i'll have to call the zotero API, which requires `api.zotero.org` to be added to connect-src for CSP

Reply:
Why? You'd call Zotero API from php not from the browser.